### ISSUE TITLE
kube-proxy: use --bind-address for iptables

### ISSUE THREAD

**Issue by thockin** [kubernetes/kubernetes#9288]
From #6910 

@biswars said:
> I have just gone through this chain and trying to find out a way to force kube-proxy to use the ip address of eth1 in the iptables
> rules setting (I understand using "publicIPs" in service is meant for external access) instead of eth0. The kubernetes set up I
> have done is completely inside the private network and eth0 is meant for NAT which should not be used for internal communication
>  within the cluster.
>
> kube-proxy has the flag "--bind-address" for binding to all interface but this definitely does not apply while setting up the iptables rule.

@thockin said:
> I think it is reasonable (needs testing) to use the --bind-address as the
> iptables target if --bind-address is specified.  Needs REALLY serious
> testing because it is subtle..

**Comment by biswars** [comment]
Thank @thockin . The suggestion of using --bind-address option worked well when I was using kubernetes 0.16 version. When I tried to set up the same using the latest version 1.0.3, I see the following error repeatedly in the kube-proxy logs. Here 10.0.2.15 is my eth0 address and I have passed the eth1 address in the --bind-address while starting the proxy (/opt/bin/kube-proxy --master=172.17.8.102:8080 --bind-address=172.17.8.103 --logtostderr=true). Any suggestion to make it work ?

Aug 21 11:48:38 core-03 kube-proxy[922]: E0821 11:48:38.722388     922 proxysocket.go:100] Dial failed: dial tcp 10.0.2.15:6443: connection refused
Aug 21 11:48:38 core-03 kube-proxy[922]: E0821 11:48:38.722480     922 proxysocket.go:100] Dial failed: dial tcp 10.0.2.15:6443: connection refused
Aug 21 11:48:38 core-03 kube-proxy[922]: E0821 11:48:38.722562     922 proxysocket.go:100] Dial failed: dial tcp 10.0.2.15:6443: connection refused
Aug 21 11:48:38 core-03 kube-proxy[922]: E0821 11:48:38.722580     922 proxysocket.go:134] Failed to connect to balancer: failed to connect to an endpoint.
Aug 21 11:48:38 core-03 kube-proxy[922]: E0821 11:48:38.888165     922 proxysocket.go:100] Dial failed: dial tcp 10.0.2.15:6443: connection refused
